nixos/modules/services/monitoring/prometheus/exporters/fritz.nix

```diff
@@ -0,0 +1,102 @@
+{ config, lib, pkgs, options }:
+let
+  inherit (lib) mkOption types mdDoc concatStringsSep;
+  cfg = config.services.prometheus.exporters.fritz;
+  yaml = pkgs.formats.yaml { };
+in
+{
+  port = 9787;
+
+  extraOpts = {
+    configFile = mkOption {
+      type = types.path;
+      default = yaml.generate "fritz-exporter.yaml" cfg.settings;
+      defaultText = "A configuration file generated from the provided nix attributes settings option.";
+      description = mdDoc ''
+        A YAML config file which can be used to configure fritz-exporter instead of the nix-generated file.
+      '';
+    };
+    settings = mkOption {
```

Comment:
This settings re-export kind of looks strange to me. i understand why you are doing it.
But i think we might could do similar to what the dmarc exporter  https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/monitoring/prometheus/exporters/dmarc.nix#L8 is doing, than re-defining the already existing options, or is there a other module doing the same already? 
For context i am looking for kind of consistency in the way the exporters are being configured and at least on the ones i am using this PR differs in having an additional "settings" inbetween.

Reply:
When I started writing this module I first tried out the same approach dmarc-exporter uses,
but this allows an easier more one-to-one mapping to the upstream config format (and also allows specifying settings which might not have an option in the module yet).

We could switch this to